src/Build/BackEnd/Components/Communications/NodeProviderOutOfProc.cs

```diff
@@ -63,10 +63,11 @@ public int AvailableNodes
         /// </summary>
         /// <param name="enableNodeReuse">Is reuse of build nodes allowed?</param>
         /// <param name="enableLowPriority">Is the build running at low priority?</param>
-        internal static Handshake GetHandshake(bool enableNodeReuse, bool enableLowPriority)
+        /// <param name="specialNode">/Indicates if node can not accept standard MSBuild work</param>
+        internal static Handshake GetHandshake(bool enableNodeReuse, bool enableLowPriority, bool specialNode)
```

Comment:
The purpose of this flag is to avoid the BuildManager trying to connect to this node similar to a build request worker node right?
Maybe it would be better to create some sort of `NodeKind` enum to make things more explicit.

Reply:
During construction of the Handshake struct, these parameters are converted to `HandshakeOptions` enum, so the behavior is already happening, just one step later.